### ISSUE TITLE
Corrupt webp output from png input

### ISSUE THREAD

**Issue by styfle** [GoogleChromeLabs/squoosh#1102]
**Describe the bug**
Converting a png to webp corrupts the image.

**To Reproduce**
1. Create a new directory, called `example`
2. Download [teamviewer.png](https://user-images.githubusercontent.com/229881/125861296-74f99605-b9a5-4259-8591-5e52f8c815a7.png) to `example/teamviewer.png`
3. Create `example/index.js` with the following code:
```js
const fs = require('fs/promises')
const { ImagePool } = require('@squoosh/lib');

async function main(input, output) {
  const imagePool = new ImagePool();
  const image = imagePool.ingestImage(input);
  await image.decoded;
  await image.preprocess({ resize: { enabled: true, width: 32, height: 32 } });
  await image.encode({ webp: {} });
  const rawEncodedImage = (await image.encodedWith.webp).binary;
  await fs.writeFile(output, rawEncodedImage);
  await imagePool.close();
  console.log(`Saved to ${output}`)
}

main('./teamviewer.png', './teamviewer.webp').catch(console.error)
```
4. Change directory to `example` and run `node index.js`

**Expected behavior**
Exampled the output file `teamviewer.webp` to look similar to the png input.

**Version:**
 - OS w/ version: MacOS 11.4
 - Node version: 14.17.3
 - npm version: 6.14.13

**Is your issue related to the quality of image compression?**
Please attach original and output images (you can drag & drop to attach). 
- Original image: [teamviewer.png](https://user-images.githubusercontent.com/229881/125861296-74f99605-b9a5-4259-8591-5e52f8c815a7.png)
- Output image from Squoosh: [teamviewer.webp.zip](https://github.com/GoogleChromeLabs/squoosh/files/6826390/output-teamviewer.webp.zip)

**Additional context, screenshots, screencasts**
Original issue https://github.com/vercel/next.js/issues/27092

Note that the squoosh.app website does not seem to be affected by the bug, just `@squoosh/lib` (currently v0.3.1).

**Comment by surma** [comment]
Quick update: Can reproduce the bug in our PNG decoder that we use in Node (on the web we can ask the browser to decode the image for us). Our decoder doesn’t handle 16-bit images, which is why this one is getting corrupted.

**Comment by surma** [comment]
Fixed in v0.4.0 of libsquoosh :)
